activesupport/test/core_ext/time_ext_test.rb

```diff
@@ -736,6 +736,18 @@ def test_future_with_time_current_as_time_with_zone
     end
   end
 
+  def test_before
+    assert_equal false, Time.utc(2017, 3, 6, 12, 0, 0).before?(Time.utc(2017, 3, 6, 11, 59, 59))
+    assert_equal false, Time.utc(2017, 3, 6, 12, 0, 0).before?(Time.utc(2017, 3, 6, 12, 0, 0))
+    assert_equal true, Time.utc(2017, 3, 6, 12, 0, 0).before?(Time.utc(2017, 3, 6, 12, 00, 1))
+  end
```

Comment:
Instead of testing behavior (since what you're really testing is the underlying implementation of `:>` and `:<` 

```
time = Time.now
time.method(:before) == time.method(:<)
time.method(:after) == time.method(:>)
```

Reply:
Interesting, I hadn't considered tests like that. Even though `before?` and `after?` are simply alias methods, I think it's valuable to have tests that look at actual dates and times since we'll always expect them to pass regardless of any changes to `<` or `>`. What do you think?

Reply:
We prefer to test behavior in Rails, not the implementation. If the implementation changes we don't have to change the test. So this test is actually nice.